.github/workflows/semver-checks.yml

```diff
@@ -0,0 +1,50 @@
+on:   # yamllint disable-line rule:truthy
+  pull_request:
+
+name: Check semver breaks
+
+jobs:
+  Prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly_version: ${{ steps.read_toolchain.outputs.nightly_version }}
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+      - name: "Read nightly version"
+        id: read_toolchain
+        run: echo "nightly_version=$(cat nightly-version)" >> $GITHUB_OUTPUT
+
+  API:
+    needs: Prepare
+    name: API - nightly toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # we need full history for cargo semver-checks
+      - name: "Select toolchain"
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: ${{ needs.Prepare.outputs.nightly_version }}
```

Comment:
I suggest that rather than having two jobs just to fill this we should just call `rustup default $(cat nightly-version)` here instead of the external action

Reply:
Oh, nice. If that works we should definitely do it.

Reply:
We could maybe fold this into `test.sh` if we added some sort of warning and found a way to put it back when we were done.

Reply:
Ok, good suggestion. I've reduced to one job in f8c91b0.
Still, I don't know how to do it in one step since the `with:` yml part cannot take the `cat` command. It is a 2-step thing:

1. install `rustup` with `dtolnay/toolchain`
2. set the `rustup default` with the output of `cat`

Reply:
> We could maybe fold this into `test.sh` if we added some sort of warning and found a way to put it back when we were done.

You mean add it to https://github.com/rust-bitcoin/rust-bitcoin-maintainer-tools/blob/master/ci/run_task.sh ?